scale/job/models.py

```diff
@@ -1049,6 +1051,77 @@ def get_exes(self, started=None, ended=None, statuses=None, job_type_ids=None, j
+        :type job_id: int
+        :param started: Query job executions updated after this amount of time.
+        :type started: :class:`datetime.datetime`
+        :param ended: Query job executions updated before this amount of time.
+        :type ended: :class:`datetime.datetime`
+        :param statuses: Query job executions with the a specific status.
+        :type statuses: [string]
+        :param node_ids: Query job executions that ran on a node with the identifier.
+        :type node_ids: [int]
+        :returns: The list of job executions that match the job identifier.
+        :rtype: [:class:`job.models.JobExecution`]
+        """
+
+        # Fetch a list of job executions
+        job_exes = JobExecution.objects.all().select_related('job', 'job_type', 'node', 'jobexecutionend')
+        job_exes = job_exes.defer('stdout', 'stderr')
+
+        # Apply job filtering
+        job_exes = job_exes.filter(job__id=job_id)
+        
+        # Apply time range filtering
+        if started:
+            job_exes = job_exes.filter(started__gte=started)
+        if ended:
+            job_exes = job_exes.filter(jobexecutionend__ended__lte=ended)
+
+        # Apply status and node filtering
+        if statuses:
+            if 'RUNNING' in statuses:
+                # This is a special case where we have to use exclusion so that running executions (no job_exe_end) are
+                # included
+                exclude_statues = []
+                for status in ['COMPLETED', 'FAILED', 'CANCELED']:
+                    if status not in statuses:
+                        exclude_statues.append(status)
+                job_exes = job_exes.exclude(jobexecutionend__status__in=exclude_statues)
+            else:
+                job_exes = job_exes.filter(jobexecutionend__status__in=statuses)
+        if node_ids:
+            job_exes = job_exes.filter(node_id__in=node_ids)
+
+        # Apply sorting
+        job_exes = job_exes.order_by('exe_num')
+
+        return job_exes
+
+    def get_job_exe_details(self, job_id, exe_num):
+        """Returns additional details about a job execution related to the given job identifier and execution number.
+
+        :param job_id: Query job executions associated with the job identifier.
+        :type job_id: int
+        :param exe_num: Query job executions associated with the execution number.
+        :type exe_num: int
+        :returns: Details about the job execution that match the job execution identifier.
+        :rtype: [:class:`job.models.JobExecution`]
+        """
+
+        # Fetch a list of job executions
+        job_exe = JobExecution.objects.all().select_related('job', 'job_type', 'node', 'jobexecutionend',
+                                                            'jobexecutionoutput')
+        job_exe = job_exe.defer('stdout', 'stderr')
+
+        # Apply job and execution filtering
+        job_exe = job_exe.filter(job__id=job_id, exe_num=exe_num)
```

Comment:
I would recommend using get() instead of filter() to guarantee that only one row matches.